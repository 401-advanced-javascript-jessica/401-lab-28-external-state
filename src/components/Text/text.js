import React from 'react';

class Text extends React.Component {
  render() {
    return (
            <div>
                <div>Name: {this.props.name} </div>
                <p># of Updates: {this.props.count} </p>
            </div>
    );
  }
}

export default Text;
