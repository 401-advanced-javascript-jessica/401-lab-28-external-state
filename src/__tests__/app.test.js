import React from 'react';
import renderer from 'react-test-renderer';
import Text from '../components/Text/text.js';
import Form from '../components/Form/form.js';
import App from '../app';

describe("<App />", () => {
    it('Text is rendered at application start', () => {
        let app = shallow(<App />);
        expect(app.find('Form').exists()).toBe(true);
        expect(app.find('Text').exists()).toBe(true);
    });

    it('App should increment', () => {
        let app = mount(<App />);
        let form = app.find('Form');
        let button = form.find('button');
        button.simulate('submit');
        expect(app.state("count")).toEqual(1);
        expect(app.find('p').text()).toEqual('# of Updates: 1 ');
    });

    it('renders properly', ()=> {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});