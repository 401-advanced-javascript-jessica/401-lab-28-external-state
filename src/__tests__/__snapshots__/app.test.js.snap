// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders properly 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value="Jessica"
    />
    <button
      type="submit"
    >
       Update | 
      0
    </button>
  </form>,
  <div>
    <div>
      Name: 
      Jessica
       
    </div>
    <p>
      # of Updates: 
      0
       
    </p>
  </div>,
]
`;
